internal/datastore/revisions/timestamprevision.go

```diff
@@ -0,0 +1,81 @@
+package revisions
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/authzed/spicedb/pkg/datastore"
+)
+
+// TimestampRevision is a revision that is a timestamp.
+type TimestampRevision int64
```

Comment:
why not uint64?

Reply:
Timestamps before supported negative numbers and I wasn't 100% certain if we needed to continue doing so, so I allowed it